src/Map.js

```diff
@@ -169,39 +169,101 @@ export function Map(props){
+        /*
+         * frm: BUG FIX: The original code (December 12, 2019) did not account for the possibility that 
+         *               all of the events might be private (and hence none of them would have a location).
+         *               The original code assumed that there was at least one event with a location, and
+         *               so there was no protection around the line that assigned a value to "lat".
+         *               But if all of the events are private, then there is no "location" on any of them
+         *               and hence dereferencing "location" on any of them will blow up.
+         *               
+         *               There was another bug too - namely that the code to set a value for the "first"
+         *               event with a valid location is an if statment rather than a loop over all of the
+         *               events.  I have replaced the if-stmt with a for loop.
+         *               
+         */
+        /*
+         * frm: original code:
+         *
         let first = 0;
         if (!('location' in props.events[first]) || !('location' in props.events[first]['location']) || !('latitude' in props.events[first]['location']['location'])) {
           first++;
         }
+         *
+         */
+
+        // Find out whether there are any events in the list that are not private
+        let first = -1;  
+        for (let i=0; i<props.events.length; i++) {
+            if (
+              ('location' in props.events[i]) && 
+              ('location' in props.events[i]['location']) && 
+              ('latitude' in props.events[i]['location']['location'])
+            ) 
+            {
+              first = i;
+              break;
+            }
+        }
 
-        var lat = props.events[first]['location']['location']['latitude'];
-        var long = props.events[first]['location']['location']['longitude'];
+        if (first !== -1) {  // There is at least one event that is not private (hence visible on map)
 
-        if(center[0] !== lat || center[0] !== long){
-          setCenter([lat, long]);
-          setNewCenter(true);
-        }
+            var lat = props.events[first]['location']['location']['latitude'];
+            var long = props.events[first]['location']['location']['longitude'];
+
+            if(center[0] !== lat || center[0] !== long){
+              setCenter([lat, long]);
+              setNewCenter(true);
+            }
 
-        var places = {};
+            var places = {};
 
-        props.events.forEach(function(event, index) {
+            props.events.forEach(function(event, index) {
 
-          //If has longitude and latitute
-          if ('location' in event && 'location' in event['location'] && 'latitude' in event['location']['location']) {
+              //If has longitude and latitute
+              if ('location' in event && 'location' in event['location'] && 'latitude' in event['location']['location']) {
```

Comment:
Suggestion: Duplicate logic from above. Might make sense to make a small fn

Reply:
Yeah - the original code had the sequential conditions repeated.  I didn't love it, but I decided to leave it to avoid noise when folks looked at a diff.  A small function, isValidLocation() would be preferable.  Happy to put that in.  Stated differently, I am sensitive to differing ideas about coding style, and I didn't want to do too much right off until I had an idea of what Jason cared about vs. what he doesn't care about.

Reply:
Fair enough! I expect either is fine :)

Reply:
Added a new function, eventHasValidLocation(), and put it in a new utility module, Util.js.  The new function is called in both EventList.js and Map.js.  I feel cleaner now ;-)